tray_windows.c

```diff
@@ -92,9 +108,78 @@ static HMENU _tray_menu(struct tray_menu *m, UINT *id) {
   return hmenu;
 }
 
+struct icon_info _create_icon_info(const char * path) {
+  struct icon_info info;
+  info.path = path;
+  ExtractIconEx(path, 0, &info.large_icon, &info.icon, 1);
+  info.notification_icon = LoadImageA(NULL, path, IMAGE_ICON, GetSystemMetrics(SM_CXICON) * 2, GetSystemMetrics(SM_CYICON) * 2, LR_LOADFROMFILE);
+  return info;
+}
+
+void _init_icon_cache(const char ** paths, int count) {
+  icon_info_count = count;
+  icon_infos = malloc(sizeof(struct icon_info) * icon_info_count);
+
+  for (int i = 0; i < count; ++i) {
+    icon_infos[i] = _create_icon_info(paths[i]);
+  }
+}
+
+void _destroy_icon_cache() {
+  for (int i = 0; i < icon_info_count; ++i) {
+    DestroyIcon(icon_infos[i].icon);
+    DestroyIcon(icon_infos[i].large_icon);
+    DestroyIcon(icon_infos[i].notification_icon);
+  }
+
+  free(icon_infos);
+  icon_infos = NULL;
+  icon_info_count = 0;
+}
+
+HICON _fetch_cached_icon(const char * path, enum IconType icon_type) {
+  for (int i = 0; i < icon_info_count; ++i) {
+    // Note: we're doing an address check here since the assumption is
+    // the icons are hardcoded and all references are to the same instance.
+    if (icon_infos[i].path == path) {
```

Comment:
This might work now, but let's use `strcmp()` to be safe in case these assumptions are broken in the future.